### ISSUE TITLE
ChangeLog for driver-only work

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#7413]
The work on driver-only ECC consists of lots of incremental changes that are either internal, or partial steps towards a larger goal. It would be tedious to announce each step in the ChangeLog, as there would be temporary restrictions that would be lifted in the next PR.

So, it's much better to put a ChangeLog entry only once things are complete enough that they're easy to describe and correspond to something users actually care about.

However, if a release is made before we reach that stage, then we need to announce and document the state at the time of the release. This issue is present in the release EPIC to make sure we don't forget.

Fixed by #8242

**Comment by mpg** [comment]
Note: at this point this is a duplicate of #7757 (except we want 7757 in the driver-only ECC EPIC where it belongs, and we wanted this one in the release EPIC to make sure we don't forget and release without it). Both are going to be fixed by the same PR, which will be raised this week.
